eng/SourceBuild.props

```diff
@@ -0,0 +1,23 @@
+<Project>
+
+  <PropertyGroup>
+    <GitHubRepositoryName>test-templates</GitHubRepositoryName>
+    <SourceBuildManagedOnly>true</SourceBuildManagedOnly>
+  </PropertyGroup>
+
+  <Target Name="ApplySourceBuildPatchFiles"
+          Condition="
+            '$(ArcadeBuildFromSource)' == 'true'"
+          AfterTargets="PrepareInnerSourceBuildRepoRoot">
+
+    <ItemGroup>
+      <SourceBuildPatchFile Include="$(RepositoryEngineeringDir)source-build-patches\*.patch" />
```

Comment:
`RepositoryEngineeringDir` refers to the original/outer repo. This can break badly, right? Is there something that that points to the `eng` dir of the inner repo?

Reply:
I can't think of anything that would happen differently depending on whether this uses the files in the outer repo in the inner repo. The outer and inner repo are always the same source code. The `PrepareInnerSourceBuildRepoRoot` target only executes in the context of the outer repo, so the patches will always be there.

I think there is a quirk where if the outer repo has some unstaged file additions, they don't transfer over properly into the inner repo. (This is based on `git stash` behavior.) But that's just dev workflow, which isn't a priority for ArPow right now--and it's easy to just stage the files to work around this.

There wouldn't be any harm in adding a prop in dotnet/arcade to refer to the inner eng dir, though: https://github.com/dotnet/arcade/blob/e4ab5cb034f0c40b4a52a5952b20809daaa2f39b/src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcade.targets#L19-L20